packages/paste-libraries/react-textarea-autosize/src/calculateNodeHeight.ts

```diff
@@ -0,0 +1,50 @@
+import type {SizingData} from './getSizingData';
+import {forceHiddenStyles} from './forceHiddenStyles';
+import {getHeight} from './getHeight';
+
+// TODO: use labelled tuples once they are avaiable:
+//   export type CalculatedNodeHeights = [height: number, rowHeight: number];
+// https://github.com/microsoft/TypeScript/issues/28259
+export type CalculatedNodeHeights = number[];
+
+export function calculateNodeHeight(
+  hiddenTextarea: HTMLTextAreaElement | null,
+  sizingData: SizingData,
+  value: string,
+  minRows = 1,
+  maxRows = 999
+): CalculatedNodeHeights {
+  const {paddingSize, borderSize, sizingStyle} = sizingData;
+  const {boxSizing} = sizingStyle;
+
+  if (hiddenTextarea == null) return [0, 0];
+
+  Object.keys(sizingStyle).forEach((_key) => {
+    const key = _key as keyof typeof sizingStyle;
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    hiddenTextarea!.style[key] = sizingStyle[key] as any;
```

Comment:
Ok! I can add a check that style is defined. 

I'm not sure I need to do the second part because don't objects work where when you have an object and you set the value of a key, it creates the key if it doesn't already exist?

Reply:
I'm going to leave this as is because it is copied from the original library.